File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    // Asks the user to input the amount of change owed to them.
    do
    {
        dollars = get_float("Change owed: ");
    }
    // Checks if the user entered a positive number, will return question if condition hasn't been met.
    while (dollars <= 0);

    // Converting currency from Dollars to Cents.
    int cents = round(dollars * 100);
    int coins = 0;

    // Checks if it can return a Quarter.
    while (cents >= 25)
    {
        cents -= 25;
        coins++;
    }
    // Checks if it can return a Dime.
    while (cents >= 10)
    {
        cents -= 10;
        coins++;
    }
    // Checks if it can return a Nickel.
    while (cents >= 5)
    {
        cents -= 5;
        coins++;
    }
    //Checks if it can return a Penny.
    while (cents >= 1)
    {
        cents -= 1;
        coins++;
    }
    // Returns a value equal to the amount of change returned.
    printf("%i\n", coins);
}

File: pset6 - Python 🐍/cash.py
from cs50 import get_float

# Asks user for input the change value, it will repeat the question if the answer isn't a positive number
dollars = 0
while dollars <= 0:
    dollars = get_float("Change owed: ")

# Converting currency from Dollars to Cents
cents = round(dollars * 100)
coins = 0

# Counts through the coins until the change owed is equal to 0
while cents > 0:
    # Quarter
    if cents >= 25:
        cents -= 25
        coins += 1
    # Dime
    elif cents >= 10:
        cents -= 10
        coins += 1
    # Nickel
    elif cents >= 5:
        cents -= 5
        coins += 1
    else:
    # Penny
        cents -= 1
        coins += 1

# Returns a value equal to the amount of change returned.
print(f"You are owed {coins} coins in total.")
